vmhost/contexts/async.go

```diff
@@ -625,6 +631,39 @@ func (context *asyncContext) addAsyncCall(groupID string, call *vmhost.AsyncCall
 	return nil
 }
 
+func (context *asyncContext) setBuiltinFlagsOnCall(call *vmhost.AsyncCall) error {
+	destination := call.Destination
+	data := call.Data
+
+	runtime := context.host.Runtime()
+
+	// If ArgParser cannot read the Data field, then this is neither a SC call,
+	// nor a built-in function call.
+	functionName, args, err := context.callArgsParser.ParseData(string(data))
+	if err != nil {
+		return err
+	}
+
+	if context.host.IsBuiltinFunctionName(functionName) {
+		call.IsBuiltinFunctionCall = true
+		vmInput := runtime.GetVMInput()
+		actualDestination := context.determineDestinationForAsyncCall(destination, data)
+		isNoCallAfterESDTTransfer, _, _ := context.isESDTTransferOnReturnDataFromFunctionAndArgs(
+			runtime.GetContextAddress(),
+			actualDestination,
+			functionName,
+			args)
+		isAsyncCall := vmInput.CallType == vm.AsynchronousCall
+		isReturningCall := bytes.Equal(vmInput.CallerAddr, actualDestination)
```

Comment:
Consider moving these 2 booleans on line 648 and if they're both false, setting `call.IsESDTOnCallback = false` and returning nil.